### PR TITLE
Migrate Linux android gallery `e2e` and `hybrid` tests to build+test

### PR DESCRIPTION
**Pull request by keyonghan**
Part of https://github.com/flutter/flutter/issues/103542.

This migrates `flutter_gallery__transition_perf_hybrid` and `flutter_gallery__transition_perf_e2e` to build+test model. Validated via led, but marking as `bringup: true` to initiate in CI.

### CODE REVIEW

**Review by CaseyHillers — APPROVED** [review]

**Review by zanderso — CHANGES_REQUESTED** [review]
Given that the recipe hit a bug today, I'd rather hold on this until the one benchmark that is currently under build_test can successfully be promoted out of bringup.

**Comment by keyonghan** [comment]
> Given that the recipe hit a bug today, I'd rather hold on this until the one benchmark that is currently under build_test can successfully be promoted out of bringup.

Target `flutter_gallery__transition_perf` is in prod, and this is unblocked.

**Comment by zanderso** [comment]
@keyonghan Since the build_test changes started landing, it looks like we haven't been getting `flutter_gallery__transition_perf` benchmark numbers reported to SkiaPerf for Moto G4 for the past two days: https://flutter-flutter-perf.skia.org/e/?begin=1661456541&end=1661992145&queries=test%3Dflutter_gallery__transition_perf&requestType=0

Could you please take a look?

**Comment by keyonghan** [comment]
Thanks @zanderso for catching this. I validated the metrics when it ran in staging, but missed it when moved to prod. Working on a patch.

**Comment by keyonghan** [comment]
Now the target `flutter_gallery__transition_perf` in running in prod, and performance metrics are being populated correctly:
https://flutter-flutter-perf.skia.org/e/?queries=test%3Dflutter_gallery__transition_perf
We should be good now /cc @zanderso

File: dev/devicelab/lib/tasks/gallery.dart [hunk 1]
```diff
@@ -25,14 +25,29 @@ TaskFunction createGalleryTransitionTest({bool semanticsEnabled = false}) {
...
+
 TaskFunction createGalleryTransitionE2ETest({
   bool semanticsEnabled = false,
   bool enableImpeller = false,
 }) {
   return GalleryTransitionTest(
-    testFile: semanticsEnabled
-        ? 'transitions_perf_e2e_with_semantics'
-        : 'transitions_perf_e2e',
+    testFile: semanticsEnabled ? 'transitions_perf_e2e_with_semantics' : 'transitions_perf_e2e',
```
Comment (zanderso) [inline dev/devicelab/lib/tasks/gallery.dart:50]: Why did the formatting change here?

File: dev/devicelab/lib/tasks/gallery.dart [hunk 2]
```diff
@@ -42,17 +57,25 @@ TaskFunction createGalleryTransitionE2ETest({
...
+  );
+}
+
 TaskFunction createGalleryTransitionHybridTest({bool semanticsEnabled = false}) {
   return GalleryTransitionTest(
     semanticsEnabled: semanticsEnabled,
-    driverFile: semanticsEnabled
-        ? 'transitions_perf_hybrid_with_semantics_test'
-        : 'transitions_perf_hybrid_test',
+    driverFile: semanticsEnabled ? 'transitions_perf_hybrid_with_semantics_test' : 'transitions_perf_hybrid_test',
```
Comment (zanderso) [inline dev/devicelab/lib/tasks/gallery.dart:74]: ditto question

File: dev/devicelab/lib/tasks/gallery.dart [hunk 3]
```diff
@@ -42,17 +57,25 @@ TaskFunction createGalleryTransitionE2ETest({
...
   );
 }
 
 class GalleryTransitionTest {
-
```
Comment (zanderso) [inline dev/devicelab/lib/tasks/gallery.dart:55]: ditto question

File: dev/devicelab/lib/tasks/gallery.dart [hunk 4]
```diff
@@ -101,31 +124,29 @@ class GalleryTransitionTest {
         applicationBinaryPath = 'build/app/outputs/flutter-apk/app-profile.apk';
       }
 
-      final String testDriver = driverFile ?? (semanticsEnabled
-          ? '${testFile}_with_semantics_test'
-          : '${testFile}_test');
+      final String testDriver =
```
Comment (zanderso) [inline dev/devicelab/lib/tasks/gallery.dart:127]: ditto question

File: dev/devicelab/lib/tasks/gallery.dart [hunk 5]
```diff
@@ -101,31 +124,29 @@ class GalleryTransitionTest {
...
+          driverFile ?? (semanticsEnabled ? '${testFile}_with_semantics_test' : '${testFile}_test');
       section('DRIVE START');
       await flutter('drive', options: <String>[
         '--no-dds',
         '--profile',
         if (enableImpeller) '--enable-impeller',
-        if (needFullTimeline)
-          '--trace-startup',
+        if (needFullTimeline) '--trace-startup',
```
Comment (zanderso) [inline dev/devicelab/lib/tasks/gallery.dart:134]: ditto question

File: dev/devicelab/lib/tasks/gallery.dart [hunk 6]
```diff
@@ -101,31 +124,29 @@ class GalleryTransitionTest {
...
         if (applicationBinaryPath != null)
           '--use-application-binary=$applicationBinaryPath'
-        else
-          ...<String>[
-            '-t',
-            'test_driver/$testFile.dart',
-          ],
+        else ...<String>[
```
Comment (zanderso) [inline dev/devicelab/lib/tasks/gallery.dart:137]: ditto question

File: dev/devicelab/lib/tasks/gallery.dart [hunk 7]
```diff
@@ -101,31 +124,29 @@ class GalleryTransitionTest {
...
+        ],
         '--driver',
         'test_driver/$testDriver.dart',
         '-d',
         deviceId,
       ]);
     });
 
-    final String testOutputDirectory = Platform.environment['FLUTTER_TEST_OUTPUTS_DIR'] ?? '${galleryDirectory.path}/build';
+    final String testOutputDirectory =
```
Comment (zanderso) [inline dev/devicelab/lib/tasks/gallery.dart:148]: New newline after `=`.

**Review by zanderso — COMMENTED** [review]
There are a bunch of unrelated formatting changes in this PR that don't seem to be in alignment with the style guide. I'd suggest backing those out.

**Review by zanderso — APPROVED** [review]

File: dev/devicelab/lib/tasks/gallery.dart (hunk 1 above)
Comment (keyonghan) [reply dev/devicelab/lib/tasks/gallery.dart]: I guess I formatted locally with 120 length.. Restored back.

Comment (zanderso) [reply dev/devicelab/lib/tasks/gallery.dart]: The `flutter/flutter` repo doesn't use the Dart formatter for formatting, so you might want to make sure you have it turned off in your IDE when working there.

**Pull request merged**